### PR TITLE
STCOR-820 For the /reset-password route, allow token to be specified in the path or query arguments

### PR DESCRIPTION
**Pull request by ryandberger**
- Fulfills [STCOR-820](https://folio-org.atlassian.net/browse/STCOR-820)
- Add support for optional URL parameter `resetToken` which allows us to use longer keys since path has a much smaller allowable length than URL params
- Removed token param passed into `<CreateResetPassword>` component since it's not used there.

### CODE REVIEW

**Review by zburke — COMMENTED** [review]
I have be honest, without any comments, I found this work hard to follow. IIUC, it used to be that CreateResetPasswordControl had a required `token` prop that's now optional, and if the prop is absent it'll look for a query-string param named `resetToken`. Is that correct, or at least partially so? 

I realize the lack of documentation did not start with you, and I'm legit impressed with your ability to parachute in and figure out how to modify this component. For those of us who are less gifted, however, in-line comments are always appreciated :)

File: src/RootWithIntl.js [hunk 1]
```diff
@@ -173,7 +173,7 @@ class RootWithIntl extends React.Component {
                       <Switch>
                         <TitledRoute
                           name="CreateResetPassword"
-                          path="/reset-password/:token"
+                          path="/reset-password/:token?"
```
[the remark this answers is not in the document]
Comment (ryandberger) [reply src/RootWithIntl.js:176]: I'll update with a comment. It's a `react-router` convention that makes that part of the path optional. In this case it allows us to accept the path `/reset-password?resetToken=ABC`.

[the remark this answers is not in the document]
Comment (zburke) [reply src/RootWithIntl.js:176]: Nevermind, I misread it as `/reset-password/token?` without the leading colon 🤦 and so thought this was something else. Sorry for the distraction.
